src/Core/Package.vala

```diff
@@ -91,6 +91,10 @@ public class AppCenterCore.Package : Object {
                 return false;
             }
 
+            if (get_newest_release ().get_urgency () == AppStream.UrgencyKind.CRITICAL) {
```

Comment:
Please check if `get_newest_release` didn't return `null` before checking it's urgency.